### DISCUSSION TITLE
Generating a static page for us without a server

### DISCUSSION THREAD

**Discussion by thefuzzy0ne** [nuxt/nuxt#17413] [Questions] [answered]
Can nuxt generate a static page for use without a server? I am trying to make an application for a friend, and I don't want to set up a web server on her machine. I'd just like for her to double-click the file so it opens in her browser and she can get on with using it. By default, nuxt prefixes all assets with a forward slash, so it's expecting to use a server root for the assets, but I would instead like for it to prefix assets with './', so it simply looks in the current directory (same one as index.html), is there any way to do this with nuxt, or am I asking too much?

Thank you.

EDIT: I have tried:
```
export default {
  build: {
	publicPath: './'	
  },
  ...
```

in my nuxt.config.js, to no avail.

EDIT 2: 
```
I have also tried
export default {
  router: {
    base: "./",
  },
  ...
```
Which sort of works (The app loads, but gives me an internally generated 404 error), but clicking on a link redirects me to the the root of my harddrive.

**Comment by Flanker72** [comment, marked as answer]
Hi! There is not a Nuxt problem, that's how file protocol works with browsers. To dynamically load any related data you need a some kind of a server.
If you want to distribute the final version of your page - you may place a server starter script into the result bundle, and your friend should start this script instead of index.html. That should do the magic)

The simpliest way is to use Python or some other one-liners (https://gist.github.com/willurd/5720255) - there is a plenty of variants of how to do this.

However, if you want to start without installing any dependencies - you may use a some kind of embedded web server. You may google `mongoose web server` or use some alternatives like https://sourceforge.net/projects/civetweb/ - they allow you to start local web server and to run your Nuxt project with server capabilities.

**Reply by thefuzzy0ne** [reply]
Sorry, I didn't realise then when I hit reply, it wasn't replying to you directly and may not have notified you. However, I think that ultimately, you're right. Nuxt is either the wrong tool for what I need to do and I should therefore just use Webpack, or I should just embed a microserver into the app, such as [JLHTTP](https://github.com/curtcox/JLHTTP).

**Comment by thefuzzy0ne** [comment]
Thanks for your reply. However, I'm certain that this is to do with the paths inserted during the build process. / evaluates to the root of the hard drive. If I could just change this to ./ (so everything is relative to the current directory) instead, I'm fairly sure this will all work. My friend is in America and I am in the UK, so it's not as if I can just pop round to install a web server. I wanted to make a simple solution that she could just extract and run, without any fluff. Surely, there must be a way to make a Nuxt app a sub-app of an existing app? Because the way that things currently are, it's as if Nuxt expects to **BE** the app and exist at the root of a web project, but essentially what I'm after is for it to co-exist as a part of an existing app. Does that make sense?

**Reply by danielroe** [reply]
You're probably looking for `app.baseURL`.

**Reply by thefuzzy0ne** [reply]
Unfortunately this did not seem to have any effect. Also, I am struggling to find the documentation for this. Please could you link to it?

**Comment by Flanker72** [comment]
You may check https://github.com/nuxt/framework/discussions/5517 - the suggested solution may help to generate one single file with all dependencies embedded.

**Reply by thefuzzy0ne** [reply]
Many thanks for this. This is sort of what I am after, but I don't really want everything in a single file. I think that ultimately, I'm trying to do something Nuxt just wasn't designed to do, so I think my options are to either embed a small web server, or to just go back to Webpack.
